### DISCUSSION TITLE
Installation fails: 'GLIBCXX_3.4.26' not found

### DISCUSSION THREAD

**Discussion by bobino-pepino** [Koenkk/zigbee2mqtt#23385] [General]
I am having problems after updating zigbee2mqtt from a previous version on my Raspberry Pi 4 Model B Rev 1.4, running Raspian 10.3.

I followed the instructions on website, but after the update when I try to run  it, I get the error message shown.

```
pi@MelHub:/opt/zigbee2mqtt $ npm start

node: /lib/arm-linux-gnueabihf/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by node)
```

Surfing around, it appears to be an node dependency so I tried a few versions of that to no effect. (v18.20.4, v20.9.0, v20.15.1)

Does anyone have an idea for a fix?

**Comment by bobino-pepino** [comment]
I decided that I'd try a completely fresh installation and did the following:

- Moved /opt/zigbee2mqtt to /opt/zigbee2mqtt_org
- Uninstalled all the versions of node using nvm uninstall
- Installed node version V20.9.0
- Followed zigbee2mqtt installation until I got to `node --version`, at which point the original dependency issue resulted
- I installed node version v18.20.4, after which I could run `node --version` without error
- I was then able to go through the installation instructions successfully up to and including running zigbbe2mqtt using `npm start`.
- I managed to get a few things to pair and it all looked good.
- I then followed the instructions for running under systemctl, which always used to work fine, but it failed. Issuing the error message:

```
Job for zigbee2mqtt.service failed because the control process exited with error code.
See "systemctl status zigbee2mqtt.service" and "journalctl -xe" for details.
```

Looking in the journal I got this

```
-- A start job for unit zigbee2mqtt.service has begun execution.
--
-- The job identifier is 11841.
Jul 21 10:03:07 MelHub systemd[1]: zigbee2mqtt.service: Main process exited, code=exited, status=1/FAILURE
-- Subject: Unit process exited
-- Defined-By: systemd
-- Support: https://www.debian.org/support
--
-- An ExecStart= process belonging to unit zigbee2mqtt.service has exited.
--
-- The process' exit code is 'exited' and its exit status is 1.
Jul 21 10:03:07 MelHub systemd[1]: zigbee2mqtt.service: Failed with result 'exit-code'.
-- Subject: Unit failed
-- Defined-By: systemd
-- Support: https://www.debian.org/support
--
-- The unit zigbee2mqtt.service has entered the 'failed' state with result 'exit-code'.
Jul 21 10:03:07 MelHub systemd[1]: Failed to start zigbee2mqtt.service.
-- Subject: A start job for unit zigbee2mqtt.service has failed
-- Defined-By: systemd
-- Support: https://www.debian.org/support
--
-- A start job for unit zigbee2mqtt.service has finished with a failure.
--
-- The job identifier is 11841 and the job result is failed.
Jul 21 10:03:12 MelHub python[356]: 10:03:12.001 Update: int_producer value: 14153 ts: 1721552592
```

Looking at the status I got this:

```
zigbee2mqtt.service
   Loaded: loaded (/etc/systemd/system/zigbee2mqtt.service; enabled; vendor preset: enabled)
   Active: activating (auto-restart) (Result: exit-code) since Sun 2024-07-21 10:04:18 BST; 5s ago
  Process: 30895 ExecStart=/usr/bin/node index.js (code=exited, status=1/FAILURE)
 Main PID: 30895 (code=exited, status=1/FAILURE)
```

I tried to run the commands in the systemctr config file manually and got the original dependency issue once more:

```
pi@MelHub:/opt/zigbee2mqtt $ cd /opt/zigbee2mqtt
pi@MelHub:/opt/zigbee2mqtt $ /usr/bin/node index.js
/usr/bin/node: /lib/arm-linux-gnueabihf/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /usr/bin/node)
```

I did a search from the root directory for any files that containing "GLIB" and found nothing.

```
cd /
sudo find . -name "*GLIB*" -ls
```

**Comment by francisp2** [comment]
Try upgrading to Bookworm.

**Reply by danielo515** [reply]
Is that the minimum debian version?

**Reply by francisp2** [reply]
I don't know if it is the minimum, but I sure don't get that error on my Pi 4. Maybe it would work on Bullseye too, but Buster is already EOL.

**Reply by bobino-pepino** [reply]
I did a from-scratch build using Bookworm and everything worked perfectly. Thanks for you help.
